flyteplugins/go/tasks/plugins/webapi/agent/plugin.go

```diff
@@ -60,6 +60,58 @@ func (p Plugin) ResourceRequirements(_ context.Context, _ webapi.TaskExecutionCo
 	return "default", p.cfg.ResourceConstraints, nil
 }
 
+type pluginContext struct {
+	webapi.TaskExecutionContext
+	resource webapi.Resource
+}
+
+func (p pluginContext) Resource() webapi.Resource {
+	return p.resource
+}
+
+func (p pluginContext) ResourceMeta() webapi.ResourceMeta {
+	return nil
+}
+
+func (p Plugin) Do(ctx context.Context, taskCtx webapi.TaskExecutionContext) (phase core.PhaseInfo, err error) {
+	// write the resource here
+	taskTemplate, err := taskCtx.TaskReader().Read(ctx)
+	if err != nil {
+		return core.PhaseInfoUndefined, err
+	}
+
+	inputs, err := taskCtx.InputReader().Get(ctx)
+	if err != nil {
+		return core.PhaseInfoUndefined, err
+	}
+
+	agent, err := getFinalAgent(taskTemplate.Type, p.cfg)
+	if err != nil {
+		return core.PhaseInfoUndefined, fmt.Errorf("failed to find agent agent with error: %v", err)
+	}
+
+	client, err := p.getClient(ctx, agent, p.connectionCache)
+	if err != nil {
+		return core.PhaseInfoUndefined, fmt.Errorf("failed to connect to agent with error: %v", err)
+	}
+
+	finalCtx, cancel := getFinalContext(ctx, "DoTask", agent)
+	defer cancel()
+
+	res, err := client.DoTask(finalCtx, &admin.DoTaskRequest{Inputs: inputs, Template: taskTemplate})
```

Comment:
I think we should send more information, similarly as:
```
res, err := client.CreateTask(finalCtx, &admin.CreateTaskRequest{Inputs: inputs, Template: taskTemplate, OutputPrefix: outputPrefix, TaskExecutionMetadata: &taskExecutionMetadata})
```

Reply:
The reason why I didn't send more information like `CreateTask` is because currently, the ChatGPT plugin doesn't need it.
After a discussion with @pingsutw, we decided to remove the unused arguments.
If necessary, I can add it back.
I will reply in 3 hours!
Thanks a lot

Reply:
@honnix We will support it now.
Will finish it in 2 hours

Reply:
Add it, please review, thanks a lot!